text/0112-server-engine.md

```diff
@@ -0,0 +1,558 @@
+
+**The `renderedCallback` lifecycle hook will not execute on the server:** When running in a browser, this hook is the first life cycle hook which gives the component author access to the rendered DOM elements. If the component were to attempt to access those APIs on the server it would result in a runtime error since the DOM APIs are not mocked.
+
+**Wire adapters will not be invoked:** The Wire protocol emits a stream of data to a component. The current protocol doesn't define a way today to indicate that the stream is done emitting new data. Because of this, the first version of SSR will not invoke any wire adapter. The protocol will need to be changed and new primitives will need to be added to LWC to make wire adapters compatible with SSR.
+
+#### The `renderComponent` API
+
+The `renderComponent` is the only new public API exposed with this proposal. This new API is only available in `@lwc/engine-server`. It renders a component and synchronously returns the rendered content. This proposal accepts 3 arguments:
+ - `name` (type: `string`) - The tag name of the component host element.
+ - `ctor` (type: `typeof LightningElement`) - The root LWC component constructor.
+ - `props` (optional, type: `{ [key string]: any}`) - An object representing the different properties set on the root component.
+
+This method returns the serialized HTML content rendered by the component. The serialization format is out of the scope of this proposal and will be covered in a different RFC. However, in the first version of the `@lwc/engine-server`, the `renderComponent` will produce an HTML string matching the [declarative shadow DOM proposal](https://github.com/mfreed7/declarative-shadow-dom/blob/master/README.md) format.
+
+#### Component authoring format
+
+The existing `lwc` package stays untouched and will be used to distribute the different versions of the engine. From the developer perspective, the experience writing a component remains identical. Since the LWC engine exposes different APIs depending on the environment, the application owner will be in charge of creating a different entry point for each environment.
+
+**`c/app/app.js`:**
+
+```js
+import { LightningElement } from 'lwc';
+
+export default class App extends LightningElement {}
+```
+
+**`client.js`:**
+
+```js
+import { createElement } from 'lwc'; // Resolves to `@lwc/engine-dom`
+import App from 'c/app';
+
+const app = createElement('c-app', { is: App });
+document.body.appendChild(app);
+```
+
+**`server.js`:**
+
+```js
+import { renderComponent } from '@lwc/engine-server'; // Resolves to `@lwc/engine-server`
```

Comment:
```suggestion
import { renderComponent } from 'lwc'; // Resolves to `@lwc/engine-server`
```
You mentioned in your presentation this was a typo, it's meant to be `lwc` automatically resolves to `@lwc/engine-server`.